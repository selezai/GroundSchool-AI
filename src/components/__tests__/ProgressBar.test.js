import React from 'react';
import renderer, { act } from 'react-test-renderer';
import ProgressBar from '../ProgressBar';

describe('ProgressBar Component', () => {
  it('renders correctly with default props', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <ProgressBar current={5} total={10} />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly at 0%', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <ProgressBar current={0} total={10} />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly at 100%', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <ProgressBar current={10} total={10} />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });

  it('renders without percentage text when showPercentage is false', () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <ProgressBar current={5} total={10} showPercentage={false} />
      ).toJSON();
    });
    expect(tree).toMatchSnapshot();
  });
});
