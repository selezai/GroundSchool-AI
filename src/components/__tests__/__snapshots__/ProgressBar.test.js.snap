// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Component renders correctly at 0% 1`] = `null`;

exports[`ProgressBar Component renders correctly at 100% 1`] = `null`;

exports[`ProgressBar Component renders correctly with default props 1`] = `null`;

exports[`ProgressBar Component renders without percentage text when showPercentage is false 1`] = `null`;
